src/components/homePage/HomePage.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react'
+
+import InfiniteScroll from 'react-infinite-scroll-component'
+
+import './HomePage.css'
+
+import BookItem from '../../interfaces/BookItem'
+import { getBooksPaged } from '../../services/book.service'
+import BookList from '../bookList/BookList'
+
+const initialPageLenght  = 12
+
+interface Props {
+  searchInput : string
+}
+
+function HomePage( { searchInput } : Props) {
+  const [ bookList, setBookList ] = useState<BookItem[]>([])
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ hasMore, setHasMore ] = useState(false)
+  const currentSearch = useRef<string>(searchInput)
+
+  const addNextPage = () => {
+    setPageNumber(pageNumber + 1)
+  }
+
+  const feachBooks = () => {
+    getBooksPaged({ PageNumber: pageNumber, PageLength: initialPageLenght, Where: [ { Field: 'Title', Value: searchInput, Operation: 2 } ] }).then((response =>
+    {
+      setHasMore(pageNumber * initialPageLenght <= response.data.TotalCount)
+      setBookList((bookList) => [ ...bookList, ...response.data.Items ])
+    })).catch((error) => console.error(error))
+  }
+
+  useEffect(() => {
+    if (currentSearch.current !== searchInput) {
```

Comment:
What is this logic used for?

Reply:
To check if the useEffect was called due to a search input change